src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -31,23 +32,32 @@ export function QueryPageContainerComponent(): JSX.Element {
     const fetchAndSetDataAndFilter = async (): Promise<void> => {
         const isolateResponse: Response = await fetch(ISOLATE_URL);
         const isolateProp: IsolateDTO = await isolateResponse.json();
-        const keyValueProps = Object.keys(isolateProp.isolates[0]) as DbKey[];
+
+        const adaptedDbIsolates: DbCollection = isolateProp.isolates.map(
+            ({ microorganism, samplingContext, matrix }) => ({
+                microorganism,
+                samplingContext,
+                matrix,
+            })
+        );
 
         const filterResponse: Response = await fetch(FILTER_URL);
         const filterProp: FilterConfigDTO = await filterResponse.json();
 
-        filterProp.filters.forEach((element, index) => {
-            if (element.id === "sContext") {
-                filterProp.filters[index].id = "samplingContext";
-                filterProp.filters[index].name = "samplingContext";
+        const adaptedFilterProp = filterProp.filters.map(filterObj => {
+            const tempFilterProp = { ...filterObj};

```

Comment:
Consider using a more safe way to map the data. So explicitly create the new objects:
tempFilterProp = {
id: filterObj.id
name: filterObj.name
...
}
Another way would be using a json validator/schema during the json parse step.
Reason: Actually you don't know what the fetched data looks like. By typing it to the DTO you only tell the compiler how the data should look like. At runtime invalid data will be also mapped to your adaptedFilterProps and your app might crash somewhere else in the code. By explicitly mapping the values here your app will throw an exception also at his place, so you now whats going wrong.

Reply:
done. 
- Change how to adapted FilterProp